usecases/node_api.py

```diff
@@ -0,0 +1,54 @@
+from typing import Optional
+
+from gateways.node_api_gateway import NodeApiGateway
+
+ADDRESS_BLACKLIST_RESPONSE = {
+    'success': False,
+    'message': 'Timeout due to too many transaction',
+}
+
+
+class NodeApi:
+    def __init__(self, node_api_gateway: Optional[NodeApiGateway] = None) -> Optional[None]:
+        self.node_api_gateway = node_api_gateway or NodeApiGateway()
+
+    def get_address_balance(self, address: str) -> Optional[dict]:
+        # check if blacklisted
+        if self.node_api_gateway.is_blacklisted_address_balance(address):
+            return ADDRESS_BLACKLIST_RESPONSE
+
+        try:
+            result = self.node_api_gateway.get_address_balance(address)
+            if result is None:
+                return None
+            return result.to_dict()
+        except Exception as ex:
+            if str(ex) == 'timeout':
+                # blacklist address
+                self.node_api_gateway.blacklist_address_balance(address)
+                return ADDRESS_BLACKLIST_RESPONSE
+            raise ex
+
+    def get_address_search(
+            self, address: str, count: int, page: Optional[str] = None,
+            hash: Optional[str] = None, token: Optional[str] = None) -> Optional[dict]:
+        # check if blacklisted
+        if self.node_api_gateway.is_blacklisted_address_search(address):
+            return ADDRESS_BLACKLIST_RESPONSE
+
+        try:
+            result = self.node_api_gateway.get_address_search(
+                    address,
+                    count,
+                    page,
+                    hash,
+                    token)
+            if result is None:
+                return None
+            return result.to_dict()
+        except Exception as ex:
```

Comment:
Can't we capture requests specific error for timeout?

Reply:
The `HathorCoreClient` raises an `Exception` with the text "timeout", instead of using the `requests` exception I prepared this `try..catch` to use a custom exception (i.e. `HathorCoreTimeout`).

I didn't create the `HathorCoreTimeout` exception yet because the custom errors PR is still in review.

Reply:
Ok. I guess we should wait until custom errors PR is merged to finish this PR and remove this global exception catch.

Reply:
It was merged and the PR has been updated.